LSPS0/README.md

```diff
@@ -391,6 +379,66 @@ SHOULD indicate this as an "unrecognized" error to the user.
 > then use to help guide the evolution of the specification, or
 > to comply with later versions of the LSPS spec.
 
+##### `-32602` Invalid Parameters Error
+
+An LSP that sends back an invalid parameter error with
+`code = -32602` MUST include a `data` object in the `error`
+response.
+
+This `data` object MUST contain at least the field `unrecognized`,
+a JSON array of strings, representing an unordered set of
+parameter names.
+These are the parameter names that are not recognized by the LSP as
+valid for the RPC method.
+
+For example, suppose the client sent this request:
+
+```JSON
+{
+  "jsonrpc": "2.0",
+  "method": "example.method_name",
+  "params": {
+    "v1_parameter": "value1",
+    "v2_parameter": "value2"
+  },
+  "id": "42"
+}
+```
+
+Suppose the LSP recognizes `example.method_name` as a valid
+method, and recognizes `v1_parameter` as a valid parameter of
+that method, but does not recognize `v2_parameter`.
+In that case, the LSP would respond with:
+
+```JSON
+{
+  "jsonrpc": "2.0",
+  "error": {
+    "code": -32602,
+    "message": "Invalid params",
+    "data": {
+      "unrecognized": ["v2_parameter"]
+    }
+  },
+  "id": "42"
+}
+```
+
+> **Rationale** Suppose there exists some other LSPS.
+> Suppose there is a newer version of this LSPS that defines a new
+> parameter `v2_parameter` for some method.
+> Suppose there is an even newer version of this LSPS that defines
+> an even newer parameter `v3_parameter` for that method.
+> If a client sends a request that includes `v2_parameter` *and*
+> `v3_parameter`, and the LSP does not yet support the latest
+> version of the LSPS, the LSP would return an "Invalid params"
+> -32602 error.
+> However, without the `unrecognized` field in the `data` object,
+> the client cannot know if the LSP does not support `v2_parameter`
+> *and* `v3_parameter`, or supports `v2_parameter` but not
+> `v3_parameter`.
+>
```

Comment:
This text seems to imply that changes in the spec must be implemented sequentially. (Version numbers)

I don't think this has to be true. 
1. Suppose we update LSPS2 to include splicing. (`v2` or `feature_splicing`)
2. Later we'll add the functionality to purchase channels on behalf of others (`v3` or `feature_buy_peer`)

An LSP-server might struggle to implement `feature_splicing` because they are waiting for their node implementation to support it. But they might be able to implement `feature_buy_peer` as soon as the spec is completed.

In terms of features you can implement `feature_buy_peer` before `feature_splicing`.
In terms of features it is weird to implement `v3` before you implement `v2`.

Yeah, I like the feature-wording much better 🤓

Reply:
ok

Reply:
done